x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/create_signals_route.ts

```diff
@@ -19,33 +19,20 @@ export const createCreateSignalsRoute: Hapi.ServerRoute = {
       options: {
         abortEarly: false,
       },
-      payload: Joi.object({
-        description: Joi.string().required(),
-        enabled: Joi.boolean().default(true),
-        filter: Joi.object(),
-        from: Joi.string().required(),
-        id: Joi.string().required(),
-        index: Joi.array().required(),
-        interval: Joi.string().default('5m'),
-        kql: Joi.string(),
-        max_signals: Joi.number().default(100),
-        name: Joi.string().required(),
-        severity: Joi.string().required(),
-        to: Joi.string().required(),
-        type: Joi.string()
-          .valid('filter', 'kql')
-          .required(),
-        references: Joi.array().default([]),
-      }).xor('filter', 'kql'),
+      payload: createSignalsSchema,
     },
   },
   async handler(request: SignalsRequest, headers) {
     const {
       description,
       enabled,
       filter,
-      kql,
       from,
+      query,
+      language,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      saved_id: savedId,
```

Comment:
I wonder if you tried `['saved_id']: savedId,` maybe??

Reply:
Prettier adds it back to this way if you try to change it so that won't work unfortunately.